### PR TITLE
sidebase V3

### LINKED ISSUE #2: Choose vue 3 UI framework
### Goal

List vue 3 ui component frameworks.

### Context

We may want to switch away from ant-design-vue. For this we should list and evaluate existing CSS frameworks. The comparison in the end should be similar to this one ORM comparison we did: https://github.com/sidestream-tech/shift-book/issues/49#issuecomment-1152344039

### Tasks

- [ ] list  shortcomings of ant-design vue
- [ ] List _answerable and understandable_ requirements in a first comment
  - answerable means: you should be able to give clear yes / no / number answers in a short form
  - understandable means: people with a base of web-development can read and understand them in a couple of words
  - see #1 for some requirements already gathered
- [ ] discuss requirements with @BracketJohn 
- [ ] Then create a table overview of exisitng frameworks that support vue3 
  - [ ] rate them in the requirements you listed above

### LINKED ISSUE #12: Adding HeadlessUI
### Describe the feature

HeadlessUI is a unstyled utility library made by the creators of TailwindCSS.
It extends TailwindCSS to allow one to add more complex animations and components. I believe that in some instances it is super helpful, as it allows you to create completely customised drop-downs, lists and modals.

Link: https://headlessui.com/

### Additional information

_No response_

### LINKED ISSUE #27: Investigate the usage of different Vue3 UI Component frameworks
**Goal**

In #2 we made a list of multiple Vue component frameworks we might want to add support for. The next step is to narrow down the selection and create separate branches running on these frameworks. This can be used to test their integration into our scaffold, their features and ease of use.

**Context**
We currently use AntDesign 3. As we also have a lot of experience with this framework, it will be left out of the investigation and be seen as a given, that we always provide at least one version with AntDesign. The frameworks I will be creating branches for in the future will be:

- PrimeVue
- Naive UI (thanks to @ManUtopiK 😊)
- Element UI

I can imagine checking more in the future, however to keep the scope of this issue down I have narrowed it to these three I plan to implement in the first wave.

### PR DESCRIPTION
**Pull request by BracketJohn**
Closes #2
Closes #12
Closes #27

This PR:
- switches from ant-design-vue to naive ui
- switches from TypeORM to Prisma
- removes the custom vite.conf.ts
- adds a smaller, custom `vitest.conf.ts`
- removes and reduces dev-related files:
  - removed: sidebase logos,
  - reduced: much of the readme docs,
  - reduced: example files to single welcome component
- switches from antfu-eslint to nuxt-eslint rules
- adds tailwind css vs code as a recommended extension
- expands testing:
  - E2E app example using nuxt-test-utils
  - unit-test-example using nuxt-test-utils

ToDos:
- [x] investigate whether we can work on #53 here -> nope, will leave it open for further PR

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -18,5 +18,10 @@
     "javascript",
     "typescript",
     "vue"
-  ]
+  ],
+  "tailwindCSS.experimental.configFile": ".nuxt/tailwind.config.cjs",
+  "files.associations": {
+    "*.ccs": "tailwindcss"
```
Comment (ineshbose) [inline .vscode/settings.json:24]: This is meant to be `*.css` I think!

Comment (BracketJohn) [reply .vscode/settings.json:24]: What do you mean? This is to improve editor-support and taken from https://tailwindcss.nuxt.dev/tailwind/editor-support

Comment (ineshbose) [reply .vscode/settings.json:24]: Editor support would be improved if the association is to CSS files - it's just misspelt here 🙂

Comment (BracketJohn) [reply .vscode/settings.json]: Ah of course - very good catch :D Fixed (:

**Pull request merged**